src/sentry/static/sentry/app/components/assistant/helper.jsx

```diff
@@ -72,27 +75,85 @@ const AssistantHelper = createReactClass({
     }
 
     return (
-      <div className="assistant-container">
-        {showDrawer ? (
-          <div className="assistant-drawer">
-            {this.state.currentGuide ? (
-              <GuideDrawer
-                guide={this.state.currentGuide}
-                step={this.state.currentStep}
-                onClose={closeGuide}
-              />
-            ) : (
-              <SupportDrawer onClose={this.handleSupportDrawerClose} />
-            )}
-          </div>
-        ) : (
-          <a onClick={this.handleDrawerOpen} className="assistant-cue">
-            {cueText}
-          </a>
+      <StyledHelper>
+        {showDrawer &&
+          this.state.currentGuide && (
+            <GuideDrawer
+              guide={this.state.currentGuide}
+              step={this.state.currentStep}
+              onClose={closeGuide}
+            />
+          )}
+
+        {showDrawer &&
+          !this.state.currentGuide && (
+            <SupportDrawer onClose={this.handleSupportDrawerClose} />
+          )}
+
+        {!showDrawer && (
+          <StyledAssistantContainer
+            onClick={this.handleDrawerOpen}
+            className="assistant-cue"
+            hasGuide={this.state.currentGuide}
+          >
+            <CueIcon hasGuide={this.state.currentGuide} />
+            <StyledCueText hasGuide={this.state.currentGuide}>{cueText}</StyledCueText>
+          </StyledAssistantContainer>
         )}
-      </div>
+      </StyledHelper>
     );
   },
 });
 
+//this globally controls the size of the component
+const StyledHelper = styled('div')`
+  font-size: 1.4rem;
+`;
+
+const StyledCueText = styled('span')`
+  width: 0px;
+  overflow: hidden;
+  opacity: 0;
+  transition: 0.2s all;
+  white-space: nowrap;
+  color: ${p => p.purpleDark};
+
+  ${p =>
+    p.hasGuide &&
+    `
+    width: auto;
+    opacity: 1;
+    margin: 0 1em 0 0.5em;
+  `};
+`;
+
+const StyledAssistantContainer = styled(AssistantContainer)`
+  display: flex;
+  align-items: center;
+  cursor: pointer;
+  max-width: 300px;
+  min-width: 0;
+  width: auto;
+
+  &:hover ${StyledCueText} {
+    ${p =>
+      !p.hasGuide &&
+      `
+      width: 6em;
+      // this is roughly long enough for the copy 'need help?'
```

Comment:
Is there a way we can build this to support translated strings of varying lengths?

Reply:
great question. i think eventually yeah—we just might need to rely on a bit of javascript and a system font to avoid getting false positives from the FOUT.

Reply:
we could also use this hack http://www.sawyerh.com/writing/css-transitioning-unknown-dimensions/ though it _seems_ like browser makers likely prohibit transitioning to an unknown width for good reason 🤔

Reply:
my intuition was right lol

from [this article](https://css-tricks.com/using-css-transitions-auto-dimensions/) 

> Why hasn't this problem been fixed at the browser level?
According to the Mozilla Developer Network docs, auto values have been intentionally excluded from the CSS transitions spec. It looks like it's been requested by a few people, but when you think about it, it makes at least a little sense that it hasn't been included. The browser process that re-calculates the sizes and positions of all elements based on their content and the way they interact with each other (known as "reflow") is expensive. If you were to transition an element into a height of auto, the browser would have to perform a reflow for every stage of that animation, to determine how all the other elements should move. This couldn't be cached or calculated in a simple way, since it doesn't know the starting and/or ending values until the moment the transition happens. This would significantly complicate the math that has to be done under the hood and probably degrade performance in a way that might not be obvious to the developer.

Reply:
but since we would be reflowing a very simple element it would probably be safe to use a hack

Reply:
i tried the `transform: scaleX(0.01)` trick and it looked hilarious and stupid. gonna skip this for now and we can address it more seriously if/when we decide to add translated guides.